packages/core-api/src/lib/AuthConnector/SamlAuthConnector.ts

```diff
@@ -0,0 +1,89 @@
+/*
+ * Copyright 2020 Spotify AB
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+import {
+  AuthProvider,
+  ProfileInfo,
+  BackstageIdentity,
+  DiscoveryApi,
+} from '../../apis/definitions';
+import { showLoginPopup } from '../loginPopup';
+
+type Options = {
+  discoveryApi: DiscoveryApi;
+  environment?: string;
+  provider: AuthProvider & { id: string };
+};
+
+export type SamlResponse = {
+  userId: string;
+  profile: ProfileInfo;
+  backstageIdentity: BackstageIdentity;
+};
+
+export class SamlAuthConnector<SamlResponse> {
```

Comment:
Think we could rename this to `DirectAuthConnector` or so? since it can be used for many other strategies than just SAML. Also thinking we should highlight in the docs that this can only be used to implement direct users interaction flows like the sign-in page, since it lacks the extra dialog layer that ensures that the popup is opened through a user action.

Reply:
👌 made the changes to the class name